game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProMatches.java

```diff
@@ -22,11 +22,15 @@
 
   public static Predicate<Territory> territoryCanLandAirUnits(final PlayerId player, final GameData data,
       final boolean isCombatMove, final List<Territory> enemyTerritories, final List<Territory> alliedTerritories) {
-    return Matches.territoryIsInList(alliedTerritories).or(
-        Matches.airCanLandOnThisAlliedNonConqueredLandTerritory(player, data)
-            .and(Matches.territoryIsPassableAndNotRestrictedAndOkByRelationships(player, data, isCombatMove, false,
-                false, true, true))
-            .and(Matches.territoryIsInList(enemyTerritories).negate()));
+    Predicate<Territory> match = Matches.airCanLandOnThisAlliedNonConqueredLandTerritory(player, data)
```

Comment:
Something to consider in the future when it comes to composing predicates, as we do in so many places...  I know we don't do this now, and it would be viewed as inconsistent, but a) statically importing predicate factory methods from `Matches`, and b) statically importing `g.s.util.Util#not()` and using it instead of calling `negate()` might make these chains of predicates easier to read:

```java
    Predicate<Territory> match = airCanLandOnThisAlliedNonConqueredLandTerritory(player, data)
        .and(territoryIsPassableAndNotRestrictedAndOkByRelationships(player, data, isCombatMove, false, false, true,
            true))
        .and(not(territoryIsInList(enemyTerritories)));
    if (!isCombatMove) {
      match = match.and(territoryIsNeutralButNotWater()
          .or(not(isTerritoryEnemyAndNotUnownedWaterOrImpassableOrRestricted(player, data))));
    }
```